src/test/pom-parser/pom-parser.test.js

```diff
@@ -107,111 +104,148 @@ describe('POMParser', () => {
-          .then((pomContent) => {
-            repository = pomContent.repositories[0];
-            assert.equal(repository.releases.enabled, 'up');
-            assert.equal(repository.releases.updatePolicy, 'Never');
-            assert.equal(repository.releases.checksumPolicy, 'gunna');
-            assert.equal(repository.snapshots.enabled, 'let');
-            assert.equal(repository.snapshots.updatePolicy, 'you');
-            assert.equal(repository.snapshots.checksumPolicy, 'down');
-  
-            emptyRSrepository = pomContent.repositories[1];
-            assert.equal(emptyRSrepository.releases.enabled, undefined);
-            assert.equal(emptyRSrepository.releases.updatePolicy, undefined);
-            assert.equal(emptyRSrepository.releases.checksumPolicy, undefined);
-            assert.equal(emptyRSrepository.snapshots.enabled, undefined);
-            assert.equal(emptyRSrepository.snapshots.updatePolicy, undefined);
-            assert.equal(emptyRSrepository.snapshots.checksumPolicy, undefined);
+
+        return pomParser.parsePOMFromString(rawContent).then(pomContent => {
+          repository = pomContent.repositories[0];
+          assert.equal(repository.releases.enabled, 'up');
+          assert.equal(repository.releases.updatePolicy, 'Never');
+          assert.equal(repository.releases.checksumPolicy, 'gunna');
+          assert.equal(repository.snapshots.enabled, 'let');
+          assert.equal(repository.snapshots.updatePolicy, 'you');
+          assert.equal(repository.snapshots.checksumPolicy, 'down');
+
+          emptyRSRepository = pomContent.repositories[1];
+          assert.equal(emptyRSRepository.releases.enabled, undefined);
+          assert.equal(emptyRSRepository.releases.updatePolicy, undefined);
+          assert.equal(emptyRSRepository.releases.checksumPolicy, undefined);
+          assert.equal(emptyRSRepository.snapshots.enabled, undefined);
+          assert.equal(emptyRSRepository.snapshots.updatePolicy, undefined);
+          assert.equal(emptyRSRepository.snapshots.checksumPolicy, undefined);
+        });
+      });
+    });
+    describe('Recursive Parent Pom Fetching', () => {
+      it('find parent pom using a user method', () => {
+        const rawContent = readFile('grandchild-fetch-pom.xml');
+
+        return pomParser
+          .parsePOMFromString(rawContent, function fetchParent(
+            parentGroupId,
+            parentArtifactId,
+            version,
+            repository
+          ) {
+            contents = readFile(`${parentArtifactId}-fetch-pom.xml`,{ encoding: 'utf8' });
+            return contents;
+          })
+          .then(pomContent => {
+            parentPOM = pomContent.parentPom;
+            assert.equal(parentPOM.groupId, 'child of the overlord');
+            assert.equal(parentPOM.artifactId, 'child');
+            assert.equal(parentPOM.version, '3.0.0');
+
+            assert.equal(parentPOM.parentPom.groupId, 'I am the overlord');
```

Comment:
We should probably validate more information about the parent.

I'm fine with having simpler tests to ensure that we're handling the recursion correctly, but we likely should add a test to ensure that all of the expected properties are getting pulled in for the parent as well.